testcontainers/tests/cli_client.rs

```diff
@@ -81,6 +86,28 @@ fn generic_image_exposed_ports() {
         .is_success());
 }
 
+#[test]
+fn generic_image_running_with_extra_hosts_added() {
+    let docker = clients::Cli::default();
+    let msg = WaitFor::message_on_stdout("server is ready");
+
+    let server_1 = GenericImage::new("simple_web_server", "latest").with_wait_for(msg.clone());
+    let node = docker.run(server_1);
+    let port = node.get_host_port_ipv4(80);
+
+    let msg = WaitFor::message_on_stdout("foo");
+    let server_2 = GenericImage::new("curlimages/curl", "latest")
+        .with_wait_for(msg.clone())
+        .with_entrypoint("curl");
+
+    // Override hosts for server_2 adding
+    // custom-host as an alias for localhost
+    let server_2 = RunnableImage::from((server_2, vec![format!("http://custom-host:{port}")]))
+        .with_add_host(("custom-host", "host-gateway"));
```

Comment:
What does `host-gateway` mean? Is that a docker thing?

Reply:
yup. it's a [special value](https://docs.docker.com/engine/reference/commandline/dockerd/#configure-host-gateway-ip) replacing a valid IP that can be used to refer to the host machine gateway IP

Reply:
Hmm, can we test this somehow differently? It appears that usually, the value here should be an `IpAddr`: https://docs.docker.com/engine/reference/commandline/image_build/#add-host

Perhaps we can make it an enum?

```rust
enum Host {
	Addr(IpAddr),
	HostGateway
}
```

Stringly-typed APIs aren't very user-friendly.

Reply:
> where would you place the enum? (i guess in `src/core/image.rs`)

Put it next to the API that uses it :)

> do you prefer i proceed keeping one single commit amending previous history or multiple commits to keep track of reviews/discussion?

Don't care really but preference for no force-pushing so I can see the diff between the reviews.

> about the test. any of these variant should cut it (and they do on my machine):

I think once we have the enum, that is good enough and you don't need to change the test. I just didn't want to have a stringly-typed API :)

Reply:
all good then 👍🏻!